### PR TITLE
Make capture groups work for swift testing without @Suite

### PR DESCRIPTION
**Pull request by bobvoorneveld**
Not all tests where captured with the new Swift testing implementation in 2.12.0, here's an update.

It's my first PR on this project, please let me know what I need to add.

### CODE REVIEW

**Comment by bobvoorneveld** [comment]
This might already be covered with this PR, let me know if this one is still needed.
https://github.com/cpisciotta/xcbeautify/pull/318

**Comment by ajkolean** [comment]
> This might already be covered with this PR, let me know if this one is still needed. #318

I am not sure this will help anything with Swift Testing. AFAIK `XCTest` uses this syntax not `SwiftTesting`.  I could be wrong, I am by no means an expert. This does allow for capture of `SomeStructThatsNotASuite/someFunctionName()` or `SomeStructThatsNotASuite.someFunctionName()`. Which if types XCTest outputs both of those formats, this would help!

**Comment by bobvoorneveld** [comment]
I've setup a sample project to show the case where this is happening:
https://github.com/bobvoorneveld/TestDemo

if you run:
```
xcodebuild test -workspace TestDemo.xcworkspace -scheme TestDemo -testPlan TestDemo
```

You get this output:

```
Testing started
Test suite 'OldTestCase' started on 'My Mac - xctest (19509)'
Test case 'OldTestCase.testExample()' passed on 'My Mac - xctest (19509)' (0.001 seconds)
Test suite 'TestStructdfdTests' started on 'My Mac - xctest (19511)'
Test suite 'TestStructTests' started on 'My Mac - xctest (19511)'
Test case 'TestStructdfdTests/exampleInPackage()' passed on 'My Mac - xctest (19511)' (0.000 seconds)
Test case 'TestStructTests/example()' passed on 'My Mac - xctest (19511)' (0.000 seconds)
Test suite 'OldTestCase' started on 'My Mac - TestDemo (19510)'
Test case 'OldTestCase.testExample()' passed on 'My Mac - TestDemo (19510)' (0.001 seconds)
Test suite 'SubFolderTestDemoTests' started on 'My Mac - TestDemo (19512)'
Test suite 'TestDemoTests' started on 'My Mac - TestDemo (19512)'
Test case 'TestDemoTests/exampleInProject()' passed on 'My Mac - TestDemo (19512)' (0.000 seconds)
Test case 'SubFolderTestDemoTests/example()' passed on 'My Mac - TestDemo (19512)' (0.000 seconds)
Test case 'OldTestCase.testPerformanceExample()' passed on 'My Mac - xctest (19509)' (0.735 seconds)
Test case 'OldTestCase.testPerformanceExample()' passed on 'My Mac - TestDemo (19510)' (0.546 seconds)
2024-10-16 17:51:23.242 xcodebuild[19448:6830107] [MT] IDETestOperationsObserverDebug: 1.522 elapsed -- Testing started completed.
2024-10-16 17:51:23.242 xcodebuild[19448:6830107] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
2024-10-16 17:51:23.242 xcodebuild[19448:6830107] [MT] IDETestOperationsObserverDebug: 1.522 sec, +1.522 sec -- end

Test session results, code coverage, and logs:
	/Users/bobvoorneveld/Library/Developer/Xcode/DerivedData/TestDemo-egofdzyzasaewcevnkwbcictndhs/Logs/Test/Test-TestDemo-2024.10.16_17-51-07-+0200.xcresult

** TEST SUCCEEDED **
```

It happens when you try to run the all the tests in parallel, not only swift testing.

**Comment by ajkolean** [comment]
Ahh I see. I also don't know what implications if any a test plan has on the scenario. Yeah definitely looks like this will be needed also. If you run `make install`  (or build it however) on your branch you can then use it to pipe the output like usually. Not necessary but if you want to see what other scenarios, I most definitely missed.

**Comment by bobvoorneveld** [comment]
Added the capture of a failing test, forgot that one in the first commit. Made sure that nested structs are also working, which seems to be the case:

```
Failing tests:
	StructInStructTests.functionInSubstructThatFailed()
	SubFolderTestDemoTests.exampleFalse()

** TEST FAILED **

Testing started
Test Suite OldTestCase started on 'My Mac - xctest (30948)'
    ✔ [OldTestCase] testExample on 'My Mac - xctest (30948)' (0.001 seconds)
Test Suite TestStructdfdTests started on 'My Mac - xctest (30950)'
Test Suite TestStructTests started on 'My Mac - xctest (30950)'
    ✔ [TestStructdfdTests] exampleInPackage on 'My Mac - xctest (30950)' (0.000 seconds)
    ✔ [TestStructTests] example on 'My Mac - xctest (30950)' (0.000 seconds)
Test Suite OldTestCase started on 'My Mac - TestDemo (30949)'
    ✔ [OldTestCase] testExample on 'My Mac - TestDemo (30949)' (0.001 seconds)
Test Suite SubFolderTestDemoTests started on 'My Mac - TestDemo (30957)'
Test Suite TestDemoTests started on 'My Mac - TestDemo (30957)'
Test Suite SubFolderTestDemoTests/StructInStructTests started on 'My Mac - TestDemo (30957)'
    ✔ [OldTestCase] testPerformanceExample on 'My Mac - xctest (30948)' (0.774 seconds)
    ✔ [SubFolderTestDemoTests] example on 'My Mac - TestDemo (30957)' (0.000 seconds)
    ✔ [TestDemoTests] exampleInProject on 'My Mac - TestDemo (30957)' (0.000 seconds)
    ✖ [SubFolderTestDemoTests] exampleFalse on 'My Mac - TestDemo (30957)' (0.000 seconds)
    ✔ [SubFolderTestDemoTests/StructInStructTests] functionInSubstruct on 'My Mac - TestDemo (30957)' (0.000 seconds)
    ✖ [SubFolderTestDemoTests/StructInStructTests] functionInSubstructThatFailed on 'My Mac - TestDemo (30957)' (0.001 seconds)
    ✔ [OldTestCase] testPerformanceExample on 'My Mac - TestDemo (30949)' (0.473 seconds)
 ```

**Comment by ajkolean** [comment]
Nice! Just curious. What happens if you run it without the test plan? My assumption is `XCTestPlan` causes all output to be treated in a similar way. Wondering if without it, run them individually and be different...

**Comment by bobvoorneveld** [comment]
Ok, created another project, only thing i've done is added a xctestcase and a swift testing struct in 2 files, then it happens to. So it seems to be the combination of the 2 testing frameworks, that this happens.

```
Testing started
Test suite 'OldTestCase' started on 'My Mac - TestDemo2 (43165)'
Test case 'OldTestCase.testExample()' failed on 'My Mac - TestDemo2 (43165)' (0.214 seconds)
Test suite 'TestDemo2Tests' started on 'My Mac - TestDemo2 (43172)'
Test case 'TestDemo2Tests/example()' failed on 'My Mac - TestDemo2 (43172)' (0.001 seconds)
Test case 'OldTestCase.testPerformanceExample()' passed on 'My Mac - TestDemo2 (43165)' (0.258 seconds)
```

```
Testing started
Test Suite OldTestCase started on 'My Mac - TestDemo2 (43418)'
    ✖ [OldTestCase] testExample on 'My Mac - TestDemo2 (43418)' (0.251 seconds)
Test Suite TestDemo2Tests started on 'My Mac - TestDemo2 (43426)'
    ✖ [TestDemo2Tests] example on 'My Mac - TestDemo2 (43426)' (0.000 seconds)
    ✔ [OldTestCase] testPerformanceExample on 'My Mac - TestDemo2 (43418)' (0.265 seconds)
```

No workspace, just run this:
```
xcodebuild test -scheme TestDemo2 |  /usr/local/bin/xcbeautify
```

File: Tests/XcbeautifyLibTests/ParserTests.swift [hunk 1]
```diff
@@ -175,4 +175,19 @@ final class ParserTests: XCTestCase {
         let captureGroup = try XCTUnwrap(parser.parse(line: input) as? SwiftTestingPassingArgumentCaptureGroup)
         XCTAssertEqual(captureGroup.numberOfArguments, 2)
     }
+
+    func testParsingLineSwiftTestingTestOutputWhenParallelTestingXCTestCases() throws {
+        let parser = Parser()
```
Comment (cpisciotta) [inline Tests/XcbeautifyLibTests/ParserTests.swift]: `parser` is already defined by the test case, so you can delete this line and use that instance.

File: Tests/XcbeautifyLibTests/ParserTests.swift [hunk 2]
```diff
@@ -175,4 +175,19 @@ final class ParserTests: XCTestCase {
...
+        XCTAssertEqual(resultSuccess.suite, "SomeStructThatsNotASuite")
+        XCTAssertEqual(resultSuccess.time, "13.060")
+        XCTAssertEqual(resultSuccess.testCase, "someFunctionName")
+
+        let resultFailed = try XCTUnwrap(parser.parse(line: "Test case 'SubFolderTestDemoTests/exampleFalse()' failed on 'My Mac - TestDemo (29692)' (0.001 seconds)") as? ParallelTestCaseFailedCaptureGroup)
+        XCTAssertEqual(resultFailed.device, "My Mac - TestDemo (29692)")
+        XCTAssertEqual(resultFailed.suite, "SubFolderTestDemoTests")
+        XCTAssertEqual(resultFailed.time, "0.001")
+        XCTAssertEqual(resultFailed.testCase, "exampleFalse")
+    }
```
Comment (cpisciotta) [inline Tests/XcbeautifyLibTests/ParserTests.swift]: Can you slightly rework this test for clarity?

- Split into two tests. One for pass and one for failure.
- Rename for readability.
- Split input into its own variable for readability.

```swift
func testParseSwiftTestingParallelTestOutputPassed() throws {
  let input = "Test case 'SomeStructThatsNotASuite/someFunctionName()' passed on 'Clone 1 of iPhone 16 Pro - xctest (38347)' (13.060 seconds)"
  let resultSuccess = try XCTUnwrap(parser.parse(line: input) as? ParallelTestCasePassedCaptureGroup)
  XCTAssertEqual(resultSuccess.device, "Clone 1 of iPhone 16 Pro - xctest (38347)")
  XCTAssertEqual(resultSuccess.suite, "SomeStructThatsNotASuite")
  XCTAssertEqual(resultSuccess.time, "13.060")
  XCTAssertEqual(resultSuccess.testCase, "someFunctionName")
}

func testParseSwiftTestingParallelTestOutputFailed() throws {
  let input = "Test case 'SubFolderTestDemoTests/exampleFalse()' failed on 'My Mac - TestDemo (29692)' (0.001 seconds)"
  let resultFailed = try XCTUnwrap(parser.parse(line: input) as? ParallelTestCaseFailedCaptureGroup)
  XCTAssertEqual(resultFailed.device, "My Mac - TestDemo (29692)")
  XCTAssertEqual(resultFailed.suite, "SubFolderTestDemoTests")
  XCTAssertEqual(resultFailed.time, "0.001")
  XCTAssertEqual(resultFailed.testCase, "exampleFalse")
}
```

**Review by cpisciotta — COMMENTED** [review]
Hey @bobvoorneveld! Thanks so much for your contribution! This LGTM, but I've left one request to rewrite the test. Great catch.

**Pull request closed without merging**